types/types.go

```diff
@@ -400,6 +400,38 @@ type Transaction struct {
 	Signatures            []TransactionSignature `json:"signatures,omitempty"`
 }
 
+// MarshalJSON implements json.Marshaller.
+//
+// json.Umarshaller is not implemented because the ID should be discarded.
+func (txn Transaction) MarshalJSON() ([]byte, error) {
+	jsonTxn := struct {
+		ID                    TransactionID          `json:"id"`
+		SiacoinInputs         []SiacoinInput         `json:"siacoinInputs,omitempty"`
+		SiacoinOutputs        []SiacoinOutput        `json:"siacoinOutputs,omitempty"`
+		FileContracts         []FileContract         `json:"fileContracts,omitempty"`
+		FileContractRevisions []FileContractRevision `json:"fileContractRevisions,omitempty"`
+		StorageProofs         []StorageProof         `json:"storageProofs,omitempty"`
+		SiafundInputs         []SiafundInput         `json:"siafundInputs,omitempty"`
+		SiafundOutputs        []SiafundOutput        `json:"siafundOutputs,omitempty"`
+		MinerFees             []Currency             `json:"minerFees,omitempty"`
+		ArbitraryData         [][]byte               `json:"arbitraryData,omitempty"`
+		Signatures            []TransactionSignature `json:"signatures,omitempty"`
```

Comment:
finally, a legitimate usecase for embedding:
```go
type jsonTxn Transaction // prevent recursion
return json.Marshal(struct{
    ID TransactionID `json:"id"`
    jsonTxn
}{txn.ID(), jsonTxn(txn)})

Reply:
Nice catch. The recursion got me and I gave up to quickly.